### PR TITLE
Steps to start, stop and force restart instances

### LINKED ISSUE #220: [1pt] Add separate scenario to simple start/stop instance systemd daemon

### PR DESCRIPTION
**Pull request by dokshina**
- Step `stop_instance` to stop and disable instance systemd service
- Step `start_instance` to start and enable instance systemd service
- Step `restart_instance_force` to restart systemd service without any conditions

Step `restart_instance` is reworked to check if the instance should be
restarted (if `needs_restart` flag is none).

Closes #220

### CODE REVIEW

File: tasks/steps/stop_instance.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+
+- name: 'Stop instance systemd service'
```
Comment (hackallcode) [inline tasks/steps/stop_instance.yml]: Stop and disable?

File: tasks/step_stop_instance.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+
+- import_tasks: 'prepare.yml'
+  tags:
+    - cartridge-instances
+    - cartridge-replicasets
+    - cartridge-config
```
Comment (hackallcode) [inline tasks/step_stop_instance.yml:7]:
```suggestion
  tags: cartridge-instances
```

**Comment by hackallcode** [comment]
Let's make description of this PR!

File: tasks/step_stop_instance.yml (hunk 2 above)
Comment (dokshina) [reply tasks/step_stop_instance.yml:7]: The same as I said in #265: I don't like the opportunity to accidentally skip prepare.

File: doc/scenario.md [hunk 3]
```diff
@@ -649,3 +650,11 @@ Output facts:
 
 - `dists_dirs_to_remove` - list of distribution directories paths that
   were removed.
+
+### stop_instance
+
+Stop instance systemd service.
```
Comment (hackallcode) [inline doc/scenario.md]: Stop and disable

File: tasks/step_stop_instance.yml (hunk 2 above)
Comment (hackallcode) [reply tasks/step_stop_instance.yml:7]: But it's impossible to accidentally skip it)

**Review by hackallcode — COMMENTED** [review]
What about tests?

**Comment by dokshina** [comment]
I'm not sure that we should test it. In fact, it's just a test that `systemd` module works.

**Comment by hackallcode** [comment]
> I'm not sure that we should test it. In fact, it's just a test that `systemd` module works.

And that we use correct variable with correct value (instance_info.systemd_service)

**Comment by hackallcode** [comment]
> > I'm not sure that we should test it. In fact, it's just a test that `systemd` module works.
> 
> And that we use correct variable with correct value (instance_info.systemd_service)

You can test that after completing this task `restart_instance` step does something

File: tasks/steps/stop_instance.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+
+- name: 'Stop and disable instance systemd service'
+  systemd:
+    name: '{{ instance_info.systemd_service }}'
+    state: stopped
+    enabled: false
+  tags: cartridge-instances
```
Comment (hackallcode) [inline tasks/steps/stop_instance.yml:8]: Let's use `when: stopped`, similarly to restart task. By default this variable can be equal to `true`

Comment (dokshina) [reply tasks/steps/stop_instance.yml:8]: No, everybody wants `restart/stop` without any conditions.

File: doc/scenario.md [hunk 5]
```diff
@@ -688,3 +691,27 @@ Input facts (set by config):
...
+
+Start and enable instance systemd service.
+
+Input facts (set by role):
+
+- `instance_info` - information for a current instance ([more details here](#role-facts-descriptions)).
+
+### restart_instance_force
+
+Restart instance systemd service without any conditions.
```
Comment (hackallcode) [inline doc/scenario.md]: and enable

File: molecule/start_stop/molecule.yml [hunk 6]
```diff
@@ -0,0 +1,63 @@
+---
```
Comment (hackallcode) [inline molecule/start_stop/molecule.yml:1]: use ../common/molecule_one_host.yml

File: tasks/steps/restart_instance.yml [hunk 7]
```diff
@@ -1,6 +1,29 @@
...
+        instance_info: '{{ instance_info }}'
+        check_package_updated: true
+        check_config_updated: true
+      register: needs_restart_res
+
+    - name: 'Set "needs_restart" fact'
+      set_fact:
+        needs_restart: '{{ needs_restart_res.fact }}'
+
+  when:
```
Comment (hackallcode) [inline tasks/steps/restart_instance.yml]: Move it in place between tags and block

File: tasks/steps/restart_instance_force.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+---
+
+- name: 'Restart instance systemd service'
```
Comment (hackallcode) [inline tasks/steps/restart_instance_force.yml]: and enable

**Review by hackallcode — DISMISSED** [review]

**Review by hackallcode — APPROVED** [review]

**Pull request merged**